src/QMCWaveFunctions/TWFPrototype.h

```diff
@@ -0,0 +1,223 @@
+   *  @param Aslice square matrices consistent with a ground state occupation.
+   *  @return Void
+   */
+  void get_gs_matrix(const std::vector<ValueMatrix_t>& A, std::vector<ValueMatrix_t>& Aslice);
+
+  /** @brief Calculates derivative of observable via Tr[M^{-1} dB - X * dM ].  Consistent with ground state occupation.
+   *
+   *  @param Minv. inverse of slater matrices for ground state occupations. 
+   *  @param X.  X=M^-1 B M^-1.  B observable matrix, and is consistent with ground state occupation.
+   *  @param dM. Target derivative of M, and is consistent with ground state occupation.
+   *  @param dB. Target derivative of B, and is consistent with ground state occupation.
+   *  @return Void
+   */
+  ValueType compute_gs_derivative(const std::vector<ValueMatrix_t>& Minv,
+                                  const std::vector<ValueMatrix_t>& X,
+                                  const std::vector<ValueMatrix_t>& dM,
+                                  const std::vector<ValueMatrix_t>& dB);
+
+  //////////////////////////////////////////////////////////////////////////////////////////////
+  //And now we just have some helper functions for doing useful math on our lists of matrices.//
+  //////////////////////////////////////////////////////////////////////////////////////////////
+
+  /** @brief Helper function that inverts all slater matrices in our species list.
+   *
+   *  @param M. List of slater matrices for each species.  These are square and consistent with some occupation.
+   *  @param Minv. The species by species list of inverted matrices from M.
+   *  @return Void.
+   */
+  void invert_M(const std::vector<ValueMatrix_t>& M, std::vector<ValueMatrix_t>& Minv);
+
+  /** @brief Helper function that inverts all slater matrices in our species list.
+   *
+   *  @param Minv. List of slater matrix inverses M^-1 for a given occupation. 
+   *  @param B. Observable auxiliary matrix for a given occupation.
+   *  @param X. M^-1*B*M^-1 is stored in this list of matrices.
+   *  @return Void. 
+   */
+  void build_X(const std::vector<ValueMatrix_t>& Minv,
+               const std::vector<ValueMatrix_t>& B,
+               std::vector<ValueMatrix_t>& X);
+
+  /** @brief Goes through a list of matrices and zeros them out.  Does this in place.
+   *
+   *  @param A. The list of matrices to be zeroed out.  After call, A is all zeros.
+   *  @return Void.
+   */
+  void wipe_matrix(std::vector<ValueMatrix_t>& A);
+  /** @brief Returns Tr(A*B).  Actually, we assume a block diagonal structure, so this is 
+   *    really Sum_i Tr(A_i*B_i), where i is the species index.
+   * 
+   *  @param A.  Vector of matrices A.  
+   *  @param B.  Vector of matrices B.
+   *  @return Value of Sum_i Tr(A_i*B_i).
+   */
+  ValueType trAB(const std::vector<ValueMatrix_t>& A, const std::vector<ValueMatrix_t>& B);
+
+
+private:
+  std::vector<IndexType> num_ptcls;
```

Comment:
This is new code it must follow the naming conventions for member variables.  This whole PR is harder to read because of this.